packages/medusa/src/utils/validators/date-transform.ts

```diff
@@ -3,3 +3,8 @@ export const transformDate = ({ value }): Date => {
     ? new Date(value)
     : new Date(Number(value) * 1000)
 }
+
+export const transformOptionalDate = (input) =>
+  !input.value && !Number.isInteger(input.value)
```

Comment:
q: why check if its not an integer?

Reply:
It's because 0 is an integer and if you really wanted to pass 1. Jan 1970 as a date but in unix time then you would just get a number rather than a date.

Probably nobody will ever do this (especially with the only use being in `starts_at` and `ends_at` for price lists), but I figured I would cover all bases now rather than when the issue arises (if ever)